presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeExecution.java

```diff
@@ -39,21 +40,26 @@
  * Tests can be running in Interactive Debugging Mode. Interactive Debugging Mode allows you to have an easier debugging
  * experience by allowing Spark side not launching its own native process, but instead communicating with an already
  * launched native process. This gives developers flexibility to hookup any IDEs or debuggers with the native process.
- * Following environment variables are needed in order to enable this mode:
+ * Following JVM argument is needed in order to enable this mode:
  *
- * - NATIVE_INTERACTIVE_DEBUG
- *   - Setting this to "true" will enable interactive debugging session. This means Spark will talk to an already
- *     launched native process instead of launching its own.
  * - NATIVE_PORT
  *   - This is the port your externally launched native process listens to. It is used to tell Spark where to send
- *     requests. If not set it is by default 7777.
+ *     requests.
  */
 public class TestPrestoSparkNativeExecution
         extends AbstractTestQueryFramework
 {
     private static final String SPARK_SHUFFLE_MANAGER = "spark.shuffle.manager";
     private static final String FALLBACK_SPARK_SHUFFLE_MANAGER = "spark.fallback.shuffle.manager";
 
+    private String prestoServerPath;
+
+    @BeforeTest
+    public void setup() {
+        prestoServerPath = System.getProperty("NATIVE_PORT") == null ? System.getProperty("PRESTO_SERVER") : "not set";
+        assertNotNull(prestoServerPath, "PRESTO_SERVER is not set in the system properties.");
```

Comment:
Do we still require this property when NATIVE_PORT is set?

In the error message, let's ask the user to add -DPRESTO_SERVER=.... to the JVM properties.

See presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestHiveQueries.java

        assertNotNull(prestoServerPath, "Native worker binary path is missing. Add -DPRESTO_SERVER=<path/to/presto_server> to your JVM arguments.");
        assertNotNull(dataDirectory, "Data directory path is missing. Add -DDATA_DIR=<path/to/data> to your JVM arguments.");

Reply:
We don't. When NATIVE_PORT is provided we set prestoServerPath to some dummy value instead of null. The check will fail only when NATIVE_PORT is not provided and PRESTO_SERVER is not provided.

Reply:
I see...

> assertNotNull(prestoServerPath, "PRESTO_SERVER is not set in the system properties.");

But then what's the point of this check? It can never fail.

It might be cleaner to make prestoServerPath as Optional and set NATIVE_EXECUTION_EXECUTABLE_PATH session property only when it is set.